accepted/2021/statics-in-interfaces/README.md

```diff
@@ -586,66 +639,49 @@ namespace System
+        where TSelf : IShiftOperators<TSelf, TOther, TResult>
     {
-        static abstract TResult operator /(TSelf left, TOther right);
+        static abstract TResult operator <<(TSelf value, TOther shiftAmount);
 
-        static abstract TResult checked operator /(TSelf left, TOther right);
+        static abstract TResult operator >>(TSelf value, TOther shiftAmount);
+
+        // Logical right shift
+        static abstract TResult operator >>>(TSelf value, TOther shiftAmount);
     }
 
-    public interface IModulusOperators<TSelf, TOther, TResult>
-        where TSelf : IModulusOperators<TSelf, TOther, TResult>
+    public interface ISubtractionOperators<TSelf, TOther, TResult>
+        where TSelf : ISubtractionOperators<TSelf, TOther, TResult>
     {
-        // The ECMA name is op_Modulus and so the behavior isn't actually "remainder" with regards to negative values
-
-        // Likewise the name IModulusOperators doesn't fit with the other names, so a better one is needed
-
-        static abstract TResult operator %(TSelf left, TOther right);
+        static abstract TResult operator -(TSelf left, TOther right);
 
-        static abstract TResult checked operator %(TSelf left, TOther right);
+        static abstract TResult operator checked -(TSelf left, TOther right);
     }
 
     public interface IUnaryNegationOperators<TSelf, TResult>
         where TSelf : IUnaryNegationOperators<TSelf, TResult>
     {
         static abstract TResult operator -(TSelf value);
 
-        static abstract TResult checked operator -(TSelf value);
+        static abstract TResult operator checked -(TSelf value);
     }
 
     public interface IUnaryPlusOperators<TSelf, TResult>
         where TSelf : IUnaryPlusOperators<TSelf, TResult>
     {
         static abstract TResult operator +(TSelf value);
-
-        static abstract TResult checked operator +(TSelf value);
-    }
-
-    public interface IBitwiseOperators<TSelf, TOther, TResult>
-        where TSelf : IBitwiseOperators<TSelf, TOther, TResult>
-    {
-        static abstract TResult operator &(TSelf left, TOther right);
-
-        static abstract TResult operator |(TSelf left, TOther right);
-
-        static abstract TResult operator ^(TSelf left, TOther right);
-
-        static abstract TResult operator ~(TSelf value);
     }
+}
 
-    public interface IShiftOperators<TSelf, TOther, TResult>
-        where TSelf : IShiftOperators<TSelf, TOther, TResult>
+namespace System.Numerics
+{
+    public interface IAdditiveIdentity<TSelf, TResult>
+        where TSelf : IAdditiveIdentity<TSelf, TResult>
     {
-        static abstract TResult operator <<(TSelf value, TOther shiftAmount);
-
-        static abstract TResult operator >>(TSelf value, TOther shiftAmount);
-
-        // Logical right shift
-        static abstract TResult operator >>>(TSelf value, TOther shiftAmount);
+        static abstract TResult AdditiveIdentity { get; }
```

Comment:
Shouldn't this be `TSelf` as opposed to `TResult`? I think it would be really weird if a number type had a static `AdditiveIdentity` property that is of a different type.

Reply:
Consider things like `DateTime + TimeSpan` where `TSelf` is `DateTime` but the additive identity is a `TimeSpan`

Reply:
I see, you're right. It's still not a result though, so the name `TResult` feels wrong. Maybe I would call this `TOther` - in `DateTime`, `TimeSpan` is the `TOther` in the operator.